src/dms/backend/EdFi.DataManagementService.Backend.OAuthService/OAuthManager.cs

```diff
@@ -0,0 +1,57 @@
+﻿// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using System.Text;
+
+namespace EdFi.DataManagementService.Backend.OAuthService;
+
+public class OAuthManager() : IOAuthManager
+{
+    public async Task GetAccessTokenAsync(HttpContext httpContext, string upstreamUri)
+    {
+        // Create HttpClient to proxy request.
+        var httpClientFactory = httpContext.RequestServices.GetRequiredService<IHttpClientFactory>();
```

Comment:
Although we do have access to `httpContext`, in order to get the headers and request body, what do you think about also letting dependency injection handling the `IHttpClientFactory`? That could probably be done at the class constructor level, along with the logger.

Reply:
Agreed! Tried adding the class constructor but the compiler complains about some Json serialization issue when the injecting the `IHttpClientFactory`:

```
2024-10-24 13:42:38.543 ERR Unknown Error - 0HN7KBDAG4JA3:00000001System.InvalidOperationException: Each parameter in the deserialization constructor on type 'EdFi.DataManagementService.Backend.OAuthService.OAuthManager' must bind to an object property or field on deserialization. Each parameter name must match with a property or field on the object. Fields are only considered when 'JsonSerializerOptions.IncludeFields' is enabled. The match can be case-insensitive.
```

I tried to avoid the pit of fixing by just calling the `GetRequiredService`, but it seems the DI with this class library / project will ultimately need to get solved.

Reply:
After refactoring, the IHttpClientFactory is now injected into the module and passed to the OAuthManager as a parameter. This not only removes the larger dependency on HttpContext, but also allows for the Module to handle some of the other parsing activites. This results in a method signature that only requires an instance of the client, an authHeader string, and the url of the service configured in AppSettings.json.